spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -153,11 +143,19 @@ public Page<TaskJobExecution> listJobExecutionsForJobWithStepCount(Pageable page
 	@Override
 	public Page<TaskJobExecution> listJobExecutionsForJobWithStepCountFilteredByJobInstanceId(
 			Pageable pageable,
-			int jobInstanceId,
-			String schemaTarget
-	) {
+			int jobInstanceId) {
 		Assert.notNull(pageable, "pageable must not be null");
-		return aggregateJobQueryDao.listJobExecutionsForJobWithStepCountFilteredByJobInstanceId(jobInstanceId, schemaTarget, pageable);
+		List <TaskJobExecution> jobExecutions =  getTaskJobExecutionsWithStepCountForList(
+			jobService.listJobExecutionsForJobWithStepCountFilteredByJobInstanceId(jobInstanceId, getPageOffset(pageable),
+				pageable.getPageSize()));
+		long total = 0;
+        try {
+            JobInstance jobInstance = jobService.getJobInstance(jobInstanceId);
+			total = jobService.getJobExecutionsForJobInstance(jobInstance.getJobName(), jobInstance.getInstanceId()).size();
+        } catch (Exception e) {
```

Comment:
Is this here because checked exceptions have been added to the called methods?

Reply:
Yes.   Batch now has checked exceptions when job instances can't be found.   I can make it more specific if you'd like.